packages/dd-trace/test/plugins/agent.js

```diff
@@ -35,59 +43,154 @@ function ciVisRequestHandler (request, response) {
   })
 }
 
-function addEnvironmentVariablesToHeaders (headers) {
-  // get all environment variables that start with "DD_"
-  const ddEnvVars = Object.entries(process.env)
-    .filter(([key]) => key.startsWith('DD_'))
-    .map(([key, value]) => `${key}=${value}`)
-
-  // add the DD environment variables to the header if any exist
-  // to send with trace to final agent destination
-  if (ddEnvVars.length > 0) {
-    headers['X-Datadog-Trace-Env-Variables'] = ddEnvVars.join(',')
+// create stub on the writer class method to update headers at time of trace send
+const sendPayloadMock = function (data, count, done) {
+  const thisValue = stubs.stubbedMethods._sendPayload.lastCall.thisValue
+  if (useTestAgent) {
+    // Update the headers with additional values
+    const headers = stubs.stubbedMethods._sendPayload.lastCall.thisValue._headers
+    addEnvironmentVariablesToHeaders(headers).then(async (reqHeaders) => {
+      stubs.stubbedMethods._sendPayload.lastCall.thisValue._headers = reqHeaders
+      // call original method
+      stubs.originalMethods._sendPayload.call(thisValue, data, count, done)
+    })
+  } else {
+    stubs.originalMethods._sendPayload.call(thisValue, data, count, done)
+  }
+}
+
+// create stub on the startSpan method to inject schema version and other tags
+const startSpanMock = function (name, { childOf, kind, meta, metrics, service, resource, type } = {}, enter = true) {
```

Comment:
Why is all this new code directly in the agent? Right now it doesn't really do expectations. Could the expectations be moved out?